### PR TITLE
Fix for inverse spec

### PR DESCRIPTION
**Pull request by neginraoof**
Fix for inverse spec and bounding input/outputs to float types

### CODE REVIEW

File: docs/Changelog.md [hunk 1]
```diff
@@ -14456,7 +14457,7 @@ This version of the operator has been available since version 12 of the default
 
 <dl>
 <dt><tt>X</tt> : T</dt>
-<dd>Input tensor</dd>
+<dd>Input tensor. Must be full-rank.</dd>
```
Comment (spandantiwari) [inline docs/Changelog.md:14460]: nit: To be consistent, let's say `The matrix must be invertible.`

File: docs/Changelog.md [hunk 2]
```diff
@@ -14445,8 +14445,9 @@ This version of the operator has been available since version 12 of the default
 
   Calculates inverse of a square matrix or batches of square matrices.
   Inverse takes one input tensor of shape `[*, M, M]`, where `*` is zero or more batch dimensions,
-  and the inner-most 2 dimensions form square matrices.
-  The output is a tensor of shape `[*, M, M]`, containing the individual inverses of all input submatrices.
+  and the inner-most 2 dimensions form square matrices. The matrices must be invertible (have full rank).
+  The behavior is undefined otherwise. The output is a tensor of shape `[*, M, M]`, containing the individual
```
Comment (spandantiwari) [inline docs/Changelog.md:14449]: Would it make sense to say more here? Some something like: `The matrices must be invertible (full-rank). The behavior where one of the matrices is not invertible is undefined. The implementation can choose to throw an error or output (garbage) results as is.`
@gramalingam @neginraoof @yuslepukhin

File: docs/Changelog.md (hunk 1 above)
Comment (gramalingam) [reply docs/Changelog.md:14460]: How about "Every matrix in the batch must be invertible"?

File: onnx/defs/math/defs.cc [hunk 3]
```diff
@@ -2297,21 +2297,32 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+             "tensor(int64)",
+             "tensor(float)",
+             "tensor(double)"},
+            "Constrain input type to 32-bit and 64-bit float and int tensors.")
+        .TypeConstraint(
+            "T1",
+            {"tensor(float)",
+             "tensor(double)"},
+            "Constrain output type to 32-bit and 64-bit float tensors.")
         .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
```
Comment (yuslepukhin) [inline onnx/defs/math/defs.cc:2326]: >ypeAndShapeInferenceFunctio [](start = 10, length = 27)

I think TypeAndShapeInferenceFunction needs to be revisited

**Comment by neginraoof** [comment]
Failures are caffe2 backend tests:
FAILED test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2Backend_opset7::test_scalar_type
FAILED test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset7::test_scalar_type
FAILED test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2Backend_opset8::test_scalar_type
FAILED test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset8::test_scalar_type

**Pull request merged**